cmd/subctl/cloud.go

```diff
@@ -0,0 +1,197 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subctl
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/admiral/pkg/util"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+	"github.com/submariner-io/cloud-prepare/pkg/aws"
+	"github.com/submariner-io/cloud-prepare/pkg/k8s"
+	"github.com/submariner-io/submariner-operator/internal/exit"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/cloud"
+	cloudaws "github.com/submariner-io/submariner-operator/pkg/cloud/aws"
+	cloudgcp "github.com/submariner-io/submariner-operator/pkg/cloud/gcp"
+	"github.com/submariner-io/submariner-operator/pkg/cloud/prepare"
+	"github.com/submariner-io/submariner-operator/pkg/reporter"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"os"
+	"path/filepath"
+)
+
+var (
+	cloudCmd = &cobra.Command{
+		Use:   "cloud",
+		Short: "Cloud operations",
+		Long:  `This command contains cloud operations related to Submariner installation.`,
+	}
+	port       cloud.Port
+	awsOptions cloudaws.Options
+	gcpOptions cloudgcp.Options
+)
+
+const (
+	DefaultNumGateways = 1
+    InfraID        = "infra-id"
+    Region         = "region"
+	ProjectID = "project-id"
+)
+
+func init() {
+	restConfigProducer.AddKubeContextFlag(cloudCmd)
+	k8sConfig, err := restConfigProducer.ForCluster()
+	exit.OnErrorWithMessage(err, "Failed to initialize a Kubernetes config")
+
+	reporter := reporter.NewCloudReporter()
+
+	cloudCmd.AddCommand(newPrepareCommand(k8sConfig, reporter))
+	rootCmd.AddCommand(cloudCmd)
+}
+
+// addAWSFlags adds basic flags needed by AWS.
+func addAWSFlags(command *cobra.Command) {
```

Comment:
I think anything specific to AWS should be in a separate file and in an `aws` sub package,.